File: src/main/java/com/radixpro/enigma/be/exceptions/DatabaseException.java
/*
 * Jan Kampherbeek, (c) 2020.
 * Enigma is open source.
 * Please check the file copyright.txt in the root of the source for further details.
 */

package com.radixpro.enigma.be.exceptions;

/**
 * General exception for all database errors.
 */
public class DatabaseException extends Exception {

   /**
    * Constructor takes message and passis it to the parent Exception.
    *
    * @param message Original message from the exception.
    */
   public DatabaseException(final String message) {
      super(message);
   }

}

File: src/main/java/com/radixpro/enigma/shared/exceptions/DatabaseException.kt
/*
 * Jan Kampherbeek, (c) 2020.
 * Enigma is open source.
 * Please check the file copyright.txt in the root of the source for further details.
 *
 */
package com.radixpro.enigma.shared.exceptions

/**
 * General exception for all database errors.
 */
class DatabaseException(message: String?) : Exception(message)